src/Servers/Kestrel/test/Interop.FunctionalTests/Http3/Http3RequestTests.cs

```diff
@@ -172,6 +173,73 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             }
         }
 
+        [ConditionalTheory]
+        [MsQuicSupported]
+        [InlineData(11)]
+        [InlineData(1024, Skip = "HttpClient issue https://github.com/dotnet/runtime/issues/56115")]
+        public async Task GET_ServerStreaming_ClientReadsPartialResponse(int clientBufferSize)
+        {
+            // Arrange
+            var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
+            var builder = CreateHostBuilder(async context =>
+            {
+                await context.Response.Body.WriteAsync(TestData);
+
+                await tcs.Task;
+
+                await context.Response.Body.WriteAsync(TestData);
+            });
+
+            using (var host = builder.Build())
+            using (var client = CreateClient())
+            {
+                await host.StartAsync();
+
+                var request = new HttpRequestMessage(HttpMethod.Get, $"https://127.0.0.1:{host.GetPort()}/");
+                request.Version = HttpVersion.Version30;
+                request.VersionPolicy = HttpVersionPolicy.RequestVersionExact;
+
+                // Act
+                var response = await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead);
+
+                // Assert
+                response.EnsureSuccessStatusCode();
+                Assert.Equal(HttpVersion.Version30, response.Version);
+
+                var responseStream = await response.Content.ReadAsStreamAsync();
+
+                var data = new List<byte>();
+                var buffer = new byte[clientBufferSize];
+                var readCount = 0;
+
+                while ((readCount = await responseStream.ReadAsync(buffer).DefaultTimeout()) != -1)
```

Comment:
What does responseStream.ReadAsync() returning -1 mean? Should this be checking for 0 and failing if we reach the end of the Stream early?

Reply:
It means I made a mistake 😄

It is never a problem because the break on length always comes before the stream returns 0 (which is what was meant to be tested).

I've updated all the tests to use existing helper methods.